### ISSUE TITLE
Implement a peer-to-peer chat capability

### ISSUE THREAD

**Issue by rsanchez-wsu** [rsanchez-wsu/RaiderPlanner#19]
The application should support the discovery of other instances of the application on the network.  The discover process could include advertising its own state along with querying the state of other instances.  The objective would be to support chatting and collaboration between students with compatible academic interests (as defined by the classes active in their instance, for example).  This idea needs considerable further development before work can begin.

**Comment by Bhulbert2096** [comment]
How would you like the peer-to-peer chat to be. Would you like it to be between just one person to one person? or a group? or both? Also I looked in the issue #22 and saw that there was talk of having the capability to see who is online. I think maybe having that would be second to creating the peer to peer chat because you could treat it just like talking with someone and they will answer when they can kinda thing. But really it is just a design choice.

**Comment by rsanchez-wsu** [comment]
@Bhulbert2096 I would say pick one and go with it.  The key is to think the design all the way through and make sure that arbitrary limitations are not introduced without a good reason.

**Comment by Bhulbert2096** [comment]
I am going to look into this for Project 2s submission

**Comment by rsanchez-wsu** [comment]
@Bhulbert2096 Before you start real work on this I would like for you to detail here or in #22 the details of what you will be implementing, how it will fit with the application architecture, what the future is for it, etc.

**Comment by Bhulbert2096** [comment]
What I am thinking is the following.
1) When the GUI for RaiderPlanner loads I want to add a function button underneath the sidebar at the very bottom.  It will say something like discussion, chat, or something of the like.
2) When the button is clicked it will load a small GUI. In this GUI it will ask for the users name, host, and port so that way the person can connect using a known port number that is within the appropriate range.
3) The user will hit the submit button and a GUI will be loaded which will look very similar to a text message format.  By hitting the submit button it will then close the previous GUI and load a new one. 
This will currently only support users sending text to and from.  Also in the future or possibly now create the capability for group chat also but time permitting. 
4)I want to look at getting this to work over the network.  If I am unable to figure it out before time runs out then it can be saved for next time.  It appears like it may be very simple and may simplify this process.
4B) I can simplify this setting the user to automatically connect using a standard port and a global host IP.  But this will connect the person to anyone using that hostname which can cause several problems. So I will most likely avoid this for now.

**Comment by rsanchez-wsu** [comment]
I suspect that this will be a rather large task.  You may want to consider breaking it into two parts.  The first part can be the GUI front end and perhaps that part that handles storing the message history to maintain state.  The second part can be the network communication back end.

**Comment by Bhulbert2096** [comment]
Yeah I believe it will be. I will definitely split it up thay way then. Should I split it up that way and do 2 seperate pull resquests or make it into 1?

**Comment by rsanchez-wsu** [comment]
@Bhulbert2096 It should be two separate pull requests.
